### ISSUE TITLE
Improve button accessibility

### ISSUE THREAD

**Issue by HaSistrunk** [RockefellerArchiveCenter/dimes#262]
## Is your feature request related to a problem? Please describe.
When using VoiceOver in Chrome and Safari:

1. Buttons that have label text that changes when pressed do not announce this change to the screenreader.
- Select all items / Deselect all items
- Add to list / Remove from list
- Download as .CSV / Downloading (this is a little different, and is fine as is in that it will be announced when button is pressed, button is disabled when downloading, and the SR announces when the download is complete).

2. "Add to list" button text is read as "A D D to list" by SR. 

## Expected behavior
1. These are basically toggle buttons, so the user should be able to interact with them in a way that indicates what pressing the button will do while they are focused on it even if they have just pressed it without having to leave and then refocus.

2. "Add to list" button should be read correctly.

## Describe the solution you'd like
1. Give each button a single aria-label that does not change, like "Add to list" and "Select all items," and then add an aria-pressed attribute to the button to indicate if it is pressed or not. This would indicate to the user that what the label indicates is either off or on. If there is an aria-label present, the SR will not read the button label. 
Downsides: Not be as clear as two descriptive labels, and can be confusing for sighted SR users.

## Describe alternatives you've considered
- Make the "Select all items" button a checkbox instead of a button with a label that reads "Select/deselect all items."

- Do something with state that I don't fully understand. See the recommendations in [this post](https://sarahmhigley.com/writing/playing-with-state/).

**Comment by HaSistrunk** [comment]
The facet show all / show less button also has this issue, but that pattern needs a little more investigation in terms of focus management and how to best handle truncated text, which I'm looking into.

Edit: see #263
